src/HermesSerializers/defautSerializer.js

```diff
@@ -15,100 +12,98 @@ module.exports =  {
             "Uint32Array",
             "Float32Array",
             "Float64Array",
+            "DataView",
         ];
-        const serializedArgs = [];
-        const transferable = [];
 
-        for (let i = 0; i < args.length; i++) {
-            if (args[i] instanceof Error) {
+        const { serializedArgs, transferable } = args.reduce((acc, argument) => {
+            if (argument instanceof Error) {
                 const obj = {
                     type: "Error",
-                    value: { name: args[i].name },
+                    value: { name: argument.name },
                 };
-                const keys = Object.getOwnPropertyNames(args[i]);
-                for (let k = 0; k < keys.length; k++) {
-                    obj.value[keys[k]] = args[i][keys[k]];
-                }
-                serializedArgs.push(obj);
-            } else if (args[i] instanceof DataView) {
-                transferable.push(args[i].buffer);
-                serializedArgs.push({
-                    type: "DataView",
-                    value: args[i].buffer,
+
+                const keys = Object.getOwnPropertyNames(argument);
+                keys.reduce((acc, key) => {
+                    acc[key] = argument[key];
+                    return acc;
+                }, obj.value);
+
+                acc.serializedArgs.push(obj);
+                return acc;
+            }
+
+            if (argument instanceof ArrayBuffer) {
+                acc.transferable.push(argument);
+                acc.serializedArgs.push({
+                    type: "__hermes__transferable__",
+                    className: "ArrayBuffer",
+                    index: acc.transferable.length - 1,
                 });
-            } else {
-                // transferable: ArrayBuffer
-                if (args[i] instanceof ArrayBuffer) {
-                    transferable.push(args[i]);
-                    args[i] = "hermes__transferable__" + transferable.length - 1;
+                return acc;
+            }
 
-                // tranferable: ImageData
-                } else if ("ImageData" in self && args[i] instanceof ImageData) {
-                    transferable.push(args[i].data.buffer);
-                    args[i] = {
-                        type: "hermes__transferable__ImageData",
-                        index: transferable.length - 1,
-                        width: args[i].width,
-                    };
-                // tranferable: TypedArray
-                } else {
-                    for (let t = 0; t < typedArray.length; t++) {
-                        if (args[i] instanceof self[typedArray[t]]) {
-                            transferable.push(args[i].buffer);
-                            args[i] = "hermes__transferable__" + transferable.length - 1;
-                            break;
-                        }
-                    }
-                }
+            if (argument instanceof ImageData && "ImageData" in self) {
```

Comment:
Can you explain the "ImageData" in self part ? 

Why only use it in this if ?

Reply:
ImageData is particular because he needs buffer and width,  when other transferable object just need buffer